state/storagePruningManager/storagePruningManager.go

```diff
@@ -57,28 +57,46 @@ func (spm *storagePruningManager) MarkForEviction(
 	log.Trace("trie hashes sizes", "newHashes", len(newHashes), "oldHashes", len(oldHashes))
 	removeDuplicatedKeys(oldHashes, newHashes)
 
-	if len(newHashes) > 0 && len(newRoot) > 0 {
-		newRoot = append(newRoot, byte(state.NewRoot))
-		err := spm.dbEvictionWaitingList.Put(newRoot, newHashes)
-		if err != nil {
-			return err
-		}
+	err := spm.markForEviction(newRoot, newHashes, state.NewRoot)
+	if err != nil {
+		return err
+	}
+
+	return spm.markForEviction(oldRoot, oldHashes, state.OldRoot)
+}
 
-		logMapWithTrace("MarkForEviction newHashes", "hash", newHashes)
+func (spm *storagePruningManager) markForEviction(
+	rootHash []byte,
+	hashes map[string]struct{},
+	identifier state.TriePruningIdentifier,
+) error {
+	if len(rootHash) == 0 || len(hashes) == 0 {
+		return nil
 	}
 
-	if len(oldHashes) > 0 && len(oldRoot) > 0 {
-		oldRoot = append(oldRoot, byte(state.OldRoot))
-		err := spm.dbEvictionWaitingList.Put(oldRoot, oldHashes)
-		if err != nil {
-			return err
-		}
+	rootHash = append(rootHash, byte(identifier))
 
-		logMapWithTrace("MarkForEviction oldHashes", "hash", oldHashes)
+	newHashesSlice := mapToByteSlice(hashes)
+	err := spm.dbEvictionWaitingList.Put(rootHash, newHashesSlice)
+	if err != nil {
+		return err
 	}
+
+	logMapWithTrace("MarkForEviction "+string(identifier), "hash", hashes)
 	return nil
 }
 
+func mapToByteSlice(hashesMap map[string]struct{}) [][]byte {
+	newHashesSlice := make([][]byte, len(hashesMap))
+	i := 0
+	for key := range hashesMap {
```

Comment:
is this needed?
can't we work with map[string]struct{}?

Reply:
Removed.